tools/reviewer/job/impl/CITask.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.job.impl;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.startupos.common.CommandLine;
+import com.google.startupos.common.FileUtils;
+import com.google.startupos.common.firestore.FirestoreClientFactory;
+import com.google.startupos.common.repo.GitRepo;
+import com.google.startupos.common.repo.GitRepoFactory;
+import com.google.startupos.tools.reviewer.ReviewerProtos.CIRequest;
+import com.google.startupos.tools.reviewer.ReviewerProtos.CIResponse;
+import com.google.startupos.tools.reviewer.ReviewerProtos.Repo;
+import com.google.startupos.tools.reviewer.job.tasks.Task;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class CITask extends FirestoreTaskBase implements Task {
+  private static FluentLogger log = FluentLogger.forEnclosingClass();
+
+  private final ReentrantLock lock = new ReentrantLock();
```

Comment:
Please document why we need a lock. What concurrency issue are we trying to prevent?

Reply:
The comment you added is fine. Resolving.